tests/pass/tree_borrows/wildcard/reborrow.rs

```diff
@@ -0,0 +1,167 @@
+
+    let ref2 = unsafe { &mut *reb_ptr };
+    let _int2 = ref2 as *mut u32 as usize;
+
+    //   ┌────────────┐
+    //   │            │
+    //   │ ptr_base*  │            *
+    //   │            │            │
+    //   └────────────┘            │
+    //                             │
+    //                             │
+    //                             ▼
+    //                      ┌────────────┐
+    //                      │            │
+    //                      │    reb     ├────────────┐
+    //                      │            │            │
+    //                      └──────┬─────┘            │
+    //                             │                  │
+    //                             │                  │
+    //                             ▼                  ▼
+    //                      ┌────────────┐     ┌────────────┐
+    //                      │            │     │            │
+    //                      │ ref1(Res)* │     │ ref2(Res)* │
+    //                      │            │     │            │
+    //                      └────────────┘     └────────────┘
+
+    // Writes either through ref1, ref2 or ptr_base, which are all exposed.
+    // Since we don't know which we do not apply any transitions to any of
+    // the references.
+    unsafe { wild.write(13) };
+
+    // We should be able to access any of the references.
+    assert_eq!(*ref2, 13);
+}
+
+/// Checks that accessing a reborrowed wildcard reference doesn't
+/// invalidate other reborrowed wildcard references, if they
+/// are also exposed.
+fn reborrow3() {
+    let mut x: u32 = 42;
+
+    let ptr_base = &mut x as *mut u32;
+    let int = ptr_base as usize;
+
+    let wild = int as *mut u32;
+
+    let reb1 = unsafe { &mut *wild };
+    let ref1 = &mut *reb1;
+    let _int = ref1 as *mut u32 as usize;
+
+    let reb2 = unsafe { &mut *wild };
+
+    //   ┌────────────┐
+    //   │            │
+    //   │ ptr_base*  │            *                  *
+    //   │            │            │                  │
+    //   └────────────┘            │                  │
+    //                             │                  │
+    //                             │                  │
+    //                             ▼                  ▼
+    //                      ┌────────────┐     ┌────────────┐
+    //                      │            │     │            │
+    //                      │ reb1(Res)  |     │ reb2(Res)  |
+    //                      │            │     │            │
+    //                      └──────┬─────┘     └────────────┘
+    //                             │
+    //                             │
+    //                             ▼
+    //                      ┌────────────┐
+    //                      │            │
+    //                      │ ref1(Res)* │
+    //                      │            │
+    //                      └────────────┘
+
+    // This is the only valid ordering these accesses can happen in.
```

Comment:
This should refer back to the comment for the test: accessing reb2 doesn't invalidate reb1 because ref1 exists. (Also, please pick more distinguishable names.)